jazz_codeq/test/test.js

```diff
@@ -0,0 +1,757 @@
+				sinon.assert.calledOnce(getMetricsStub);
+				sinon.assert.calledOnce(missingFieldsStub);
+				sinon.assert.calledOnce(fromDateStub);
+				sinon.assert.calledOnce(toDateStub);
+				sinon.assert.calledOnce(fromAfterToStub);
+				sinon.assert.calledOnce(getMetricsStub);
+				sinon.assert.calledOnce(sonarTokenStub);
+				sinon.assert.calledOnce(jazzTokenStub);
+				sinon.assert.calledOnce(projectBranchStub);
+				sinon.assert.calledOnce(codeqReportStub);
+
+				getQueryStub.restore();
+				getMetricsStub.restore();
+				missingFieldsStub.restore();
+				fromDateStub.restore();
+				toDateStub.restore();
+				fromAfterToStub.restore();
+				getMetricsStub.restore();
+				sonarTokenStub.restore();
+				jazzTokenStub.restore();
+				projectBranchStub.restore();
+				codeqReportStub.restore();
+			});
+		});
+	});
+
+	describe('getResponseForHelpPathString tests', () => {
+		let sandbox;
+		let event = {};
+		let config = {};
+		const data = { "pathString": "help" };
+
+		beforeEach(() => {
+			sandbox = sinon.createSandbox();
+			event = {
+				resourcePath: 'help',
+				method: 'GET',
+				stage: 'test',
+				query: {}
+			};
+			config = {
+				"METRICS_HELP": {
+					"security": {
+						"description": "Number of security vulnerabilities",
+						"unit": "count",
+						"minValue": "0",
+						"maxValue": "NA"
+					},
+					"code-coverage": {
+						"description": "Coverage answers the following question: 'Has each boolean expression been evaluated both to true and false?'. This is the density of possible branches in flow control structures that have been followed during unit tests execution.",
+						"unit": "percentage",
+						"minValue": "0",
+						"maxValue": "100"
+					}
+				}
+			};
+		});
+
+		afterEach(() => {
+			sandbox.restore();
+			event = {};
+			config = {};
+		});
+
+		it('should return error when getMetrics is called for pathString "help"', () => {
+
+			const getQueryStub = sinon.stub(utils, "getQuery");
+			const getMetricsStub = sinon.stub(utils, "getMetrics")
+				.returns({error: "Error getting metrics"});
+			
+			const result = index.getResponseForHelpPathString(event, config);
+			
+			sinon.assert.calledOnce(getQueryStub);
+			sinon.assert.calledOnce(getMetricsStub);
+			
+			expect(result.error).to.eq("Error getting metrics");
+			
+			getQueryStub.restore();
+			getMetricsStub.restore();
+		});
+
+		it('should return data when getMetrics is called for pathString "help"', () => {
+			const query = {};
+			
+			const getQueryStub = sinon.stub(utils, "getQuery").returns(query);
+			const getMetricsStub = sinon.stub(utils, "getMetrics").returns({ metrics: ['code-coverage'] });
+			
+			const results = index.getResponseForHelpPathString(event, config);
+			
+			sinon.assert.calledOnce(getQueryStub);
+			sinon.assert.calledOnce(getMetricsStub);
+				
+			expect(results.data.metrics[0].name).to.eq("code-coverage");
+			
+			getQueryStub.restore();
+			getMetricsStub.restore();
+		});
+	});
+

```

Comment:
Can you add unit test when the pathString is unsupported (neither `codeq` or `help`)?